cartographer/io/points_batch.h

```diff
@@ -51,7 +51,17 @@ struct PointsBatch {
   // Trajectory index that produced this point.
   int trajectory_index;
 
+  // The geometry of the points in a metric frame.
   std::vector<Eigen::Vector3f> points;
+
+  // Intensities are optional (might be empty). They represent the by device
```

Comment:
Intensities are optional and may be unspecified. The meaning of these intensity values varies by device. For example, the VLP16 provides values in the range [0, 100] for non-specular return values and values up to 255 for specular returns. On the other hand, Hokuyo lasers provide a 16-bit value that rarely peaks above 4096.